masq/src/terminal/terminal_interface.rs

```diff
@@ -1,28 +1,29 @@
 // Copyright (c) 2019-2021, MASQ (https://masq.ai) and/or its affiliates. All rights reserved.
 
-use crate::line_reader::{TerminalEvent, TerminalReal};
-use linefeed::{Interface, ReadResult, Writer};
+use crate::terminal::line_reader::{TerminalEvent, TerminalReal};
+use crate::terminal::secondary_infrastructure::{InterfaceWrapper, MasqTerminal, WriterLock};
+use linefeed::{Interface, Signal};
+use masq_lib::command::StdStreams;
 use masq_lib::constants::MASQ_PROMPT;
+use masq_lib::utils::WrapResult;
 use std::sync::Arc;
 
 #[cfg(not(test))]
 mod prod_cfg {
-    pub use crate::line_reader::IntegrationTestTerminal;
+    pub use crate::terminal::integration_tests_utils::IntegrationTestTerminal;
     pub use linefeed::DefaultTerminal;
 }
 
 #[cfg(test)]
 mod test_cfg {
+    pub use crate::terminal::terminal_interface::tests::result_wrapper_for_in_memory_terminal;
     pub use linefeed::memory::MemoryTerminal;
     pub use masq_lib::intentionally_blank;
 }
 
-//I'm using the system stdout handles for writing which has been a standard way in the project for a long time;
-//so instead of writing into linefeed's Writer directly I'm making use of just its locking functionality
-
-pub const MASQ_TEST_INTEGRATION_KEY: &str = "MASQ_TEST_INTEGRATION";
-pub const MASQ_TEST_INTEGRATION_VALUE: &str = "3aad217a9b9fa6d41487aef22bf678b1aee3282d884eeb\
-74b2eac7b8a3be8xzt";
+//Not correspondingly to the normal way of an implementation of linefeed, I keep using the system stdout handles for writing instead of its native writers
+//because the former is more traditional and in our case it serves its purpose as well.
```

Comment:
Not corresponding to the standard way of an implementation of line_feed, I stick with using the system stdout handles for writing in instead of the native handles provided from line_feed.